### PR TITLE
Add moar template tag tests!

### PR DESCRIPTION
**Pull request by Camwyn**
Add allow_posts function, fix typos. Tweak some comments in `event-tickets/src/template-tags/tickets.php` to make them clearer for users.
Add tests for `tribe_tickets_get_capacity`, `tribe_tickets_delete_capacity`, `tribe_tickets_update_capacity`, `tribe_tickets_get_readable_amount`, and `tribe_tickets_has_meta_fields`.

🎫 https://central.tri.be/issues/126229

### CODE REVIEW

File: src/template-tags/tickets.php [hunk 1]
```diff
@@ -445,7 +445,7 @@ function tribe_tickets_get_ticket_stock_message( Tribe__Tickets__Ticket_Object $
 
 		$sold_label = __( 'issued', 'event-tickets' );
 		if ( 'Tribe__Tickets__RSVP' === $ticket->provider_class ) {
-			$sold_label = _x( 'RSVP\'d Going', 'separate going and remain RSVPs', 'event-tickets' );
+			$sold_label = _x( "RSVP'd Going", 'separate going and remain RSVPs', 'event-tickets' );
```
Comment (Camwyn) [inline src/template-tags/tickets.php]: I feel bad about the translation thrash, but then again I feel good about not passing the `\` through to the translators?

File: src/template-tags/tickets.php [hunk 2]
```diff
@@ -748,11 +748,15 @@ function tribe_tickets_has_meta_fields( $id, $context = null ) {
 }
 
 /**
- * Updates a given Object Capacity
+ * Removes all meta for a given object capacity
+ * object can be a ticket, or an event/post with tickets
+ *
+ * Note, you can pass an event/post to this function and it will merrily change the meta values
+ * for the event - not fot the tickets!
```
Comment (Camwyn) [inline src/template-tags/tickets.php]: This part is confusing to me - what is the purpose here, did we store capacity on events at some point? Do we now?

File: tests/wpunit/Tribe/Tickets/Template_TagsTest.php [hunk 3]
```diff
@@ -42,12 +42,17 @@ public function tearDown() {
 		parent::tearDown();
 	}
 
+	protected function allow_posts() {
+		tribe_update_option( 'ticket-enabled-post-types', [
```
Comment (Camwyn) [inline tests/wpunit/Tribe/Tickets/Template_TagsTest.php]: Convenience function

File: src/template-tags/tickets.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/template-tags/tickets.php]: 👍

File: src/template-tags/tickets.php [hunk 4]
```diff
@@ -748,11 +748,15 @@ function tribe_tickets_has_meta_fields( $id, $context = null ) {
 }
 
 /**
- * Updates a given Object Capacity
+ * Removes all meta for a given object capacity
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * Removes all meta for a given object capacity. Object can be a ticket, or an event/post with tickets.
```

File: src/template-tags/tickets.php [hunk 5]
```diff
@@ -748,11 +748,15 @@ function tribe_tickets_has_meta_fields( $id, $context = null ) {
...
+ * object can be a ticket, or an event/post with tickets
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
```

File: src/template-tags/tickets.php (hunk 2 above)
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * for the event - not for the tickets!
```

Comment (sc0ttkclark) [reply src/template-tags/tickets.php]:
```suggestion
 * for the event - not for the tickets!
```

File: src/template-tags/tickets.php [hunk 6]
```diff
@@ -748,11 +748,15 @@ function tribe_tickets_has_meta_fields( $id, $context = null ) {
...
+ *
+ * Note, you can pass an event/post to this function and it will merrily change the meta values
+ * for the event - not fot the tickets!
  *
  * @since  4.6.2
  *
- * @param  int  $object  Post We are trying to save capacity
+ * @param  int|WP_Post  $object  WP_Post (or ID of post) We are trying to delete capacity from
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * @param int|WP_Post $object WP_Post (or ID of post) We are trying to delete capacity from.
```

File: src/template-tags/tickets.php [hunk 7]
```diff
@@ -789,10 +793,13 @@ function tribe_tickets_delete_capacity( $object ) {
 /**
  * Updates a given Object Capacity
  *
+ * Note, you can pass an event/post to this function and it will merrily change the meta values
+ * for the event - not fot the tickets!
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * for the event - not for the tickets!
```

File: src/template-tags/tickets.php [hunk 8]
```diff
@@ -789,10 +793,13 @@ function tribe_tickets_delete_capacity( $object ) {
...
+ *
  * @since  4.6.2
  *
  * @param  int|WP_Post|Tribe__Tickets__Ticket_Object  $object   Post We are trying to save capacity
- * @param  int  $capacty  How much we are trying to update the capacity to
+ * @param  int  $capacity  What much we are trying to update the capacity to.
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * @param int                                        $capacty How much we are trying to update the capacity to	 * @param  int  $capacity  What much we are trying to update the capacity to.
```

File: src/template-tags/tickets.php [hunk 9]
```diff
@@ -789,10 +793,13 @@ function tribe_tickets_delete_capacity( $object ) {
 /**
  * Updates a given Object Capacity
  *
+ * Note, you can pass an event/post to this function and it will merrily change the meta values
+ * for the event - not fot the tickets!
+ *
  * @since  4.6.2
  *
  * @param  int|WP_Post|Tribe__Tickets__Ticket_Object  $object   Post We are trying to save capacity
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * @param int|WP_Post|Tribe__Tickets__Ticket_Object  $object  Post We are trying to save capacity
```

File: src/template-tags/tickets.php [hunk 10]
```diff
@@ -816,9 +823,12 @@ function tribe_tickets_update_capacity( $object, $capacity ) {
 /**
  * Returns the capacity for a given Post
  *
+ * Note while we can send a post/event we do not store capacity on events
+ * so the return values will always be null
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * so the return values will always be null.
```

File: src/template-tags/tickets.php [hunk 11]
```diff
@@ -816,9 +823,12 @@ function tribe_tickets_update_capacity( $object, $capacity ) {
...
+ *
  * @since  4.6
  *
- * @param  int  $post  Post We are trying to fetch capacity
+ * @param  int|WP_Post $post Post We are trying to fetch capacity for
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * @param int|WP_Post $post Post we are trying to fetch capacity for.
```

File: src/template-tags/tickets.php [hunk 12]
```diff
@@ -884,7 +894,7 @@ function tribe_tickets_get_capacity( $post ) {
  *
  * @since  4.6
  *
- * @param  string|int $number Which you are tring to convert
+ * @param  string|int $number Which you are trying to convert
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * @param  string|int $number Which you are trying to convert.
```

File: src/template-tags/tickets.php [hunk 13]
```diff
@@ -884,7 +894,7 @@ function tribe_tickets_get_capacity( $post ) {
...
  * @param  string     $mode   Mode this post is on
```
Comment (sc0ttkclark) [inline src/template-tags/tickets.php]:
```suggestion
 * @param string     $mode   Mode this post is on.
```

File: tests/wpunit/Tribe/Tickets/Template_TagsTest.php (hunk 3 above)
Comment (sc0ttkclark) [reply tests/wpunit/Tribe/Tickets/Template_TagsTest.php]: 👍

File: src/template-tags/tickets.php (hunk 2 above)
Comment (Camwyn) [reply src/template-tags/tickets.php]: doh 🤦‍♂️

Comment (sc0ttkclark) [reply src/template-tags/tickets.php]: Intentional https://tribe.slack.com/archives/C0258NY1C/p1556119457000700

But needs more tests as per your note

**Comment by Camwyn** [comment]
Notes: I'd rather see this get merged and do the shared capacity tests later ( perhaps as part of https://central.tri.be/issues/127167 )
Also, failing Travis tests are all acceptance tests 🤦‍♂

**Comment by sc0ttkclark** [comment]
Sounds good!

**Pull request merged**